mempool/clist_mempool_test.go

```diff
@@ -267,13 +292,33 @@ func TestMempoolUpdate(t *testing.T) {
 	}
 }
 
+func TestMempoolFetchLanesInfo(t *testing.T) {
+	appInfo := abci.InfoResponse{}
+
+	_, err := FetchLanesInfo([]uint32{}, types.Lane(0))
+	require.NoError(t, err)
+
+	_, err = FetchLanesInfo(appInfo.LanePriorities, types.Lane(1))
+	require.ErrorAs(t, err, &ErrEmptyLanesDefaultLaneSet{})
+
+	_, err = FetchLanesInfo([]uint32{1}, types.Lane(0))
+	require.ErrorAs(t, err, &ErrBadDefaultLaneNonEmptyLaneList{})
+
+	_, err = FetchLanesInfo([]uint32{1, 3, 4}, types.Lane(5))
+	require.ErrorAs(t, err, &ErrDefaultLaneNotInList{})
+
+	_, err = FetchLanesInfo([]uint32{1, 3, 4, 4}, types.Lane(4))
+	require.ErrorAs(t, err, &ErrRepeatedLanes{})
+}
+
 // Test dropping CheckTx requests when rechecking transactions. It mocks an asynchronous connection
 // to the app.
 func TestMempoolUpdateDoesNotPanicWhenApplicationMissedTx(t *testing.T) {
 	mockClient := new(abciclimocks.Client)
 	mockClient.On("Start").Return(nil)
 	mockClient.On("SetLogger", mock.Anything)
 	mockClient.On("Error").Return(nil).Times(4)
+	mockClient.On("Info", mock.Anything, mock.Anything).Return(&abci.InfoResponse{}, nil)
```

Comment:
Is there a reason why are we mocking "empty" here and "one single lane" in the calls below? (just trying to understand)

Reply:
Actually, all these tests pass even when mocking `Info` with an "empty" response. There's no need that `Info` returns an lane information because it's not needed for these tests. It will be different on the PR where we implement lanes, because the test will need lane information to create the lanes to be tested. So, in summary, in this PR we should mock all `Info` calls with an empty response.